README.md

```diff
@@ -5,94 +5,140 @@
-
-## Web Flow for Authentication
+# SAP BTP Security Services Integration Libraries
+This repository offers a comprehensive set of libraries designed to simplify the integration of [SAP Business Technology Platform](https://www.sap.com/products/technology-platform.html) (SAP BTP) security services (XSUAA and Identity Services).
+Tailored to support Java EE and Spring Boot applications running on Cloud Foundry or Kubernetes environments.
+The libraries focus on streamlining [OAuth 2.0](https://oauth.net) access token validation for tokens issued by XSUAA and Identity Services, in addition, it offers a token-client library to easily fetch tokens without cumbersome setup for http requests. 
+
+## Table of Contents
+1. [Prerequisites](#prerequisites)
+2. [Usage](#usage)
+   - [2.1 Token Validation](#21-token-validation)
+     - [2.1.1 Java EE web applications](#211-Java-EE-web-applications)
+     - [2.1.2 Spring Boot applications](#212-spring-boot-web-applications)
+   - [2.2 Token Flows](#22-token-flows)
+3. [Installation](#installation)
+4. [Troubleshooting](#troubleshooting)
+5. [Contributing](#contributing)
+6. [License](#license)
+
+## Prerequisites
+Before you can use the SAP Cloud Security Services Integration libraries, you must fulfil the following requirements:
+
+1. Access to an SAP BTP account and the XSUAA or identity service.
+2. Familiarity with OAuth 2.0 and JWT (JSON Web Tokens).
+3. Knowledge of Java programming and Spring Boot framework.
+4. Java 17
+5. Maven 3.9.0 or later
+6. (Optional) Spring Boot 3.0.0 or later, Spring Security 6.0.0 or later if using the Spring integration
+
+
+## Usage
 Typical UI5 applications consist of a server providing the HTML content and one or more application serving REST APIs used by the application. Web application use the OAuth Authorization Code Flow for interactive authentication:
 1. A user accesses the web application using a browser or mobile device
-1. The web application (in typical SAP Cloud Platform applications, this is an application router) acts as OAuth client and redirects to the OAuth server for authorization
-1. Upon authentication, the web application uses the code issued by the authorization server to request an access token
-1. The web application uses the access token to request data from the OAuth resource server. The OAuth resource server validates the token using online or offline validation.
+2. The web application (in typical SAP Business Technology Platform scenario, this is an application router) acts as OAuth client and redirects requests to the OAuth server for authorization
+3. Upon authentication, the web application uses the code issued by the authorization server to request an access token
```

Comment:
Previous sentence ends with "for authorization". This one begins with "Upon authentication". Could be changed to be more accurate.